src/core/simd.d

```diff
@@ -621,6 +630,130 @@ version (D_SIMD)
             return cast(V)__simd(XMM.LODDQU, *cast(const void16*)p);
     }
 
+    /*************************************
+    * Store vector to unaligned address.
+    * This is a compiler intrinsic.
+    * Params:
+    *    p = pointer to vector
+    *    value = value to store
+    * Returns:
+    *    value
+    */
+
+    V storeUnaligned(V)(V* p, V value)
+        if (is(V == void16) ||
+            is(V == byte16) ||
+            is(V == ubyte16) ||
+            is(V == short8) ||
+            is(V == ushort8) ||
+            is(V == int4) ||
+            is(V == uint4) ||
+            is(V == long2) ||
+            is(V == ulong2) ||
+            is(V == double2) ||
+            is(V == float4))
+    {
+        pragma(inline, true);
+        static if (is(V == double2))
+            return cast(V)__simd_sto(XMM.STOUPD, *cast(void16*)p, value);
+        else static if (is(V == float4))
+            return cast(V)__simd_sto(XMM.STOUPS, *cast(void16*)p, value);
+        else
+            return cast(V)__simd_sto(XMM.STODQU, *cast(void16*)p, value);
+    }
+}
+else version (LDC)
+{
+    /**
+    Load unaligned vector from address.
+    Params:
+        p = pointer to vector
+    Returns:
+        Vector
+    */
+    pragma(inline, true);
+    V loadUnaligned(V)(const V* p)
+        if (is(V == void16) ||
+            is(V == byte16) ||
+            is(V == ubyte16) ||
+            is(V == short8) ||
+            is(V == ushort8) ||
+            is(V == int4) ||
+            is(V == uint4) ||
+            is(V == long2) ||
+            is(V == ulong2))
```

Comment:
Fulfilling the DMD interface doesn't mean we cannot add all vector types and get rid of these 128-bit vectors limitation.

Reply:
I guess you mean what I added. Although it starts to look a little too much. I think this can be done with some kind of `isVector` compile-time function.